apply_neighbourhood.json

```diff
@@ -43,7 +43,7 @@
         },
         {
             "name": "size",
-            "description": "Neighbourhood sizes along each dimension.\nThis object maps dimension names to either a physical measure, (e.g. 100 m, 10 days), or pixels (e.g. 32 pixels).\nFor dimensions not specified, the default is to provide all values. This may result in an exception if the dimension is too large to be fully included.",
+            "description": "Neighbourhood sizes along each dimension.\nThis object maps dimension names to either a physical measure, (e.g. 100 m, 10 days), or pixels (e.g. 32 pixels).\nFor dimensions not specified, the default is to provide all values. This may result in an exception if the dimension is too large to be fully included.\n\n The moving window moves with multiples of 'size' along each dimension. All values in the window specified by size can be modified by the process.\n\n The neighbourhood size should be kept small enough, to avoid running beyond computational resources, but a too small size will result in a larger number of process invocations, which may slow down processing.",
```

Comment:
I don't feel this really helps a user:

> The neighbourhood size should be kept small enough, to avoid running beyond computational resources, but a too small size will result in a larger number of process invocations, which may slow down processing.

There are no number to refer to so it's really not so meaningful what actually is too large or too small. Thus, I'm not sure it should be included here.

Reply:
Indeed, there are no numbers, maybe I can add some for better guidance. In my experience, users do appreciate this kind of advice, as otherwise they might use entirely wrong settings. For backends, it also gives an idea of what is common.
Proposal:
Window sizes for spatial dimensions typically are in the range of 64px to 512px, while overlaps of 8px to 32px are common.

Reply:
Sounds good.